### PR TITLE
Cache of component ids do not contains anymore key and uuid, use TreeRootHolder to get them

### PR DESCRIPTION
**Pull request by julienlancelot**
Cache of component ids do not contains anymore key and uuid, use TreeRootHolder to get them

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/component/TreeRootHolderImpl.java [hunk 1]
```diff
@@ -19,24 +19,53 @@
...
+    feedComponentsByRef();
   }
 
   @Override
   public Component getRoot() {
+    checkRoot();
+    return this.root;
+  }
+
+  public Component getComponentByRef(int ref) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/component/TreeRootHolderImpl.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 major

**Comment by sns-seb** [comment]
some refactoring on `DBIdsRepository` would be nice and some commit tailoring too

watch out, Travis Job is failing because of unit test `MetricDaoTest.get_nullable_by_key`

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 new issues:
* 3 major

Details:
* [Rename "item" which hides the field declared at line 60.](https://github.com/SonarSource/sonarqube/blob/32e482d22715107305e2d0b817bac48409bebcac/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L156) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* [Add the "@Override" annotation above this method signature](https://github.com/SonarSource/sonarqube/blob/32e482d22715107305e2d0b817bac48409bebcac/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L155) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)
* [Add the "@Override" annotation above this method signature](https://github.com/SonarSource/sonarqube/blob/32e482d22715107305e2d0b817bac48409bebcac/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L150) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

**Comment by julienlancelot** [comment]
Feedback applied !

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 new issues:
* 3 major

Details:
* [Add the "@Override" annotation above this method signature](https://github.com/SonarSource/sonarqube/blob/8639e535b969dab9049e184621d7eca52c1e1f46/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L155) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)
* [Add the "@Override" annotation above this method signature](https://github.com/SonarSource/sonarqube/blob/8639e535b969dab9049e184621d7eca52c1e1f46/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L150) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)
* [Rename "item" which hides the field declared at line 60.](https://github.com/SonarSource/sonarqube/blob/8639e535b969dab9049e184621d7eca52c1e1f46/server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java#L156) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by sns-seb** [comment]
you should have a short commit message (70 chars max), a line return and then the description of your commit

**Comment by julienlancelot** [comment]
Which commit are you talking about ?

**Comment by sns-seb** [comment]
all of them except https://github.com/SonarSource/sonarqube/commit/d78d3592ffb2d6d9848cf8547594a03b7a8e3451

**Comment by julienlancelot** [comment]
Is it better now ?

**Comment by sns-seb** [comment]
otherwise, LGTM

**Comment by julienlancelot** [comment]
Cool, thanks for your feedback !

**Pull request merged**